articles/quickstart/spa/vuejs/02-calling-an-api.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Calling an API
+description: This tutorial demonstrates how to make calls to an external API
+budicon: 546
+topics:
+  - quickstarts
+  - spa
+  - vuejs
+  - apis
+github:
+  path: 02-Calling-an-API
+sample_download_required_data:
+  - client
+  - api
+contentType: tutorial
+useCase: quickstart
+---
+
+<!-- markdownlint-disable MD002 MD041 -->
+
+Most single-page apps use resources from data APIs. You may want to restrict these resources so only authenticated users with sufficient privileges can access them. Auth0 lets you manage access to these resources using [API Authorization](/api-auth).
+
+In this tutorial, you'll create a simple API using [Express](https://expressjs.com) to validate incoming JSON Web Tokens. You will then see how to call this API using an access token granted by the Auth0 authorization server.
+
+<%= include('../_includes/_calling_api_create_api') %>
+
+<%= include('../_includes/_calling_api_create_backend.md') %>
+
+Finally, modify `package.json` to add two new scripts `dev` and `api` that can be used to start the frontend and the backend API together:
+
+```json
+{
+  "name": "03-calling-an-api",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "serve": "vue-cli-service serve",
+    "build": "vue-cli-service build",
+    "lint": "vue-cli-service lint",
+    "dev": "npm-run-all --parallel serve api",
+    "api": "node server.js"
+  },
+
+  // .. package dependencies and other JSON nodes
+}
+```
+
+You can now start the project using `npm run dev` in the terminal, and the frontend Vue.js application will start up alongside the backend API.
+
+### Set up a proxy to the backend API
+
+To call the API from the frontend application, the development server must be configured to proxy requests through to the backend API. To do this, add a `vue.config.js` file to the root of the project and populate it with the following code:
```

Comment:
@joshcanhelp It's a tricky one that I have constantly battled with. It's not essential to showing how Auth0 works, no. If we take this out, we'd probably have to take out everything to do with creating the backend server.

I'd be in favour of doing this as it would condense the quickstart down further to just the essentials while still showing how to _call_ an API. We could leave the actual backend in the samples as an example for the reader to view.

The only issue then is that the other SPA quickstarts have this backend. Options:

1) Leave the backend documentation where it is for consistency with other SPA quickstarts
2) Remove it in favour of making the quickstart more concise

Personally I'm leaning towards 2. We could try it and see what kind of feedback we get on it, if any?

Reply:
Now I'm split :)

My default decision is to remove anything not related to the task at hand. If it does not improve the experience, increase learning, or reduce overall complexity, I don't think it should be there. If, however, this increases the chance of success for some reason (smooth out gaps in a specific platform or something), then that's a positive. 

I'll leave it up to you but it _seems_ like this just makes a nice URL for the API which, in this case, is not relevant for the task at hand.

Reply:
It's more to make the API accessible on the same origin. When considering the bigger picture, it makes configuration and deployment a bit easier too.

I think it's more a question of whether the docs needs to detail how the sample app is built, or leave it out of the docs and have readers study the sample app for this kind of detail.

For now, I'm going to leave it as it is, as this is detail is also in the other SPA quickstarts and we haven't had any feedback on it to make us think it's detrimental to the reader's experieance. Howver, I do hope that this is something that the current work into quickstart principals can help definitively answer.

Reply:
> It's more to make the API accessible on the same origin.

Ahhhh, that didn't occur to me. Makes sense to me. Thank you for the thorough explanation.